src/lib/util/stats.ts

```diff
@@ -42,21 +42,20 @@ const detectType = (text: string): string => {
 		.split(' ')[0];
 };
 
-// manual debounce
-let timeout: number;
 export const saveStatistics = (graph: string): void => {
+	const graphType = detectType(graph);
+	console.debug(`ga: send event: render ${graphType}`);
+	logEvent('render', { graphType });
+};
+
+// manual debounce to only send analytics event every 5 seconds if same event is repeated frequently.
+const timeouts: Record<string, number> = {};
+export const logEvent = (name: string, data?: unknown): void => {
 	if (analytics) {
-		clearTimeout(timeout);
-		// Only save statistics after a 5 sec delay
-		timeout = window.setTimeout(() => {
-			const graphType = detectType(graph);
-			console.debug(`ga: send event: render ${graphType}`);
-			void logEvent('render', { graphType });
+		const key = data ? JSON.stringify({ name, data }) : name;
+		clearInterval(timeouts[key]);
+		timeouts[key] = window.setTimeout(() => {
```

Comment:
Again, I'm not sure we need the `window.` here.

```suggestion
		timeouts[key] = setTimeout(() => {
```

Reply:
Closed due to https://github.com/mermaid-js/mermaid-live-editor/pull/1031#discussion_r967667531
Seems to a be a TypeScript problem (it thinks `setTimeout()` is the Node.JS timeout, not the browser `setTimeout()`).